feat: port over protocol/agents page from v2 (#41)
hyperlane-xyz/v3-docs: docs/protocol/agents/validators.mdx

# Validators

> Validators are responsible for observing the [Mailbox](../mailbox.mdx) contract and, if required, signing its merkle root to facilitating message transmission to remote chains.

Reviewer: ```suggestion
Validators are responsible for observing the [Mailbox](../mailbox.mdx) contract and, if required, signing a merkle root to facilitate message transmission to remote chains.
```

> Unlike many other protocols, Hyperlane does **not** have an enshrined validator set. Anyone is free to run a validator and help contribute to securing interchain messages.

Reviewer 2: The last sentence is a bit misleading because the validator set registered on an ISM _is_ permissioned. People can run their own validators only as long as their receiver contract specifies a multisig ISM that does include that validator's address

> Unlike many other protocols, Hyperlane does **not** have an enshrined validator set. Anyone is free to run their own validator, as long as the receiver contract specifies a [Multisig ISM](../ISM/multisig-ISM.mdx) includes their validator.

Reviewer: ```suggestion
Unlike many other protocols, Hyperlane does **not** have an enshrined validator set. Anyone is free to run their own validator, as long as the receiver contract specifies a [Multisig ISM](../ISM/multisig-ISM.mdx) that includes their validator.
```

:::info
Want to run a validator? Follow the [validators guide](../../operate/validators/run-validators.mdx).
:::

> For messages using a [Multisig ISM](../../reference/ISM/multisig-ISM-interface.mdx), validators read the current merkle root by calling `MerkleTreeHook.latestCheckpoint()`.  Once a root has achieved sufficient [finality](https://medium.com/mechanism-labs/finality-in-blockchain-consensus-d1f83c120a9a), validators are expected to sign it and post their signature to highly available storage so that it can be aggregated by [relayers](./relayers.mdx).

Reviewer 2: here too, maybe instead of "aggregated by" we could say "collected by"

```solidity
/**
  * @notice Returns the latest checkpoint for validators to sign.
  * @return root Latest checkpointed root
  * @return index Latest checkpointed index
  */
function latestCheckpoint()
  external
  view
  returns (bytes32 root, uint256 index);
```

Validators that sign anything other than a finalized checkpoint risk compromising the safety of the protocol. If they are participating in proof of stake, this may result in their stake being slashed.

> Hyperlane is developing the validator as a Rust binary that can be easily run by anyone. Operationally, validators need to run this binary and a node for the chain that they are validating for. We hope that the majority of Hyperlane validators will come from each chain's existing node operator community.

Reviewer 2: Was this paragraph in the docs already? Validators don't _need_ to run a chain node, since they can just use an rpc provider to send their validator announce tx and read from the mailbox / merkle tree hook

Author: yes it was, might have been some conflation from polygon specific comments relating to the archive node

Author: ```suggestion
Hyperlane is developing the validator as a binary implemented in Rust that can be easily run by anyone. Operationally, validators need to run this binary and have access to an RPC provider or node for the chain that they are validating for. We hope that the majority of Hyperlane validators will come from each chain's existing node operator community.
```
updating to
